sympy/core/parameters.py

```diff
@@ -67,8 +67,7 @@ def __setattr__(self, name, value):
 
 global_parameters = _global_parameters(evaluate=True, distribute=True, exp_is_pow=False)
 
-@contextmanager
-def evaluate(x):
+class evaluate:
```

Comment:
I don't know much about contexts, but these look like welcome changes. I looked over everything and this looks good.

Reply:
I won't hurt to wait to see if anyone else has comments, otherwise this is +1 from me.